core/lib/instrument.c

```diff
@@ -6766,6 +6766,125 @@ dr_redirect_native_target(void *drcontext)
  * *loging not safe if not owning thread?
  */
 
+DR_API
+/* Schedules a shared fragment to be deleted. User must redirect control upon return
+ * using \p dr_redirect_execution() similar to what one would do when using
+ * \p dr_flush_region().
```

Comment:
What guarantees are made regarding how quickly all other threads will A) exit this fragment; B) not re-enter this fragment?

Reply:
The idea was to synch threads, make the fragment unreachable by unlinking incomings and outgoings, then add to the list for lazy deletion.

Reply:
Right, I'm requesting that the documentation make some kind of statement about not being purely synchronous, like dr_unlink_flush_region().

Reply:
Done